### DISCUSSION TITLE
help with DSDT Patching, Asrock Fatal1ty Z170

### DISCUSSION THREAD

**Discussion by ketchup48** [xCuri0/ReBarUEFI#160] [Q&A]
I ask for help with DSDT Patching,

Motherboard: Asrock Fatal1ty Z170 Gaming K4,
CPU: CoffeeTime 0.99 Mod QTJ1,
GPU: Saphire 7900 XT.

Adding FFS module – ok
Enabling hidden 4G decoding – ok
UEFI Patching - ok (Remove <64GB BAR size limit (Skylake/Kaby Lake/Coffee Lake))
CSM-Disabled

Problems: 
Above 4GB MMIO BIOS assignment - Enabled, causes a very long Windows boot time of 3-4 minutes.
GPU-Z says ReBar is Enabled, AMD Adrenalin says ReBar is not available.
In games, the FPS of 15 frames is catastrophically low, WINDOWS is "viscous".

I'm starting to do DSDT Patching, the problem is Decompiling

```
 iASL Warning: There were 14 external control methods found during
 disassembly, but only 0 were resolved (14 unresolved). Additional
 ACPI tables may be required to properly disassemble the code. This
 resulting disassembler output file may not compile because the
 disassembler did not know how many arguments to assign to the
 unresolved methods. Note: SSDTs can be dynamically loaded at
 runtime and may or may not be available via the host OS.

 To specify the tables needed to resolve external control method
 references, the -e option can be used to specify the filenames.
 Example iASL invocations:
     iasl -e ssdt1.aml ssdt2.aml ssdt3.aml -d dsdt.aml
     iasl -e dsdt.aml ssdt2.aml -d ssdt1.aml
     iasl -e ssdt*.aml -d dsdt.aml

 In addition, the -fe option can be used to specify a file containing
 control method external declarations with the associated method
 argument counts. Each line of the file must be of the form:
     External (<method pathname>, MethodObj, <argument count>)
 Invocation:
     iasl -fe refs.txt -d dsdt.aml
```

If ignored, I can't compile the file back without even making changes to the file. It is not compiled.  Errors appear.

```
Error 6126 - Compiler aborting due to parser-detected syntax error(s)
DSDT.dsl 13965: Arg1
Error 6126 - ^ syntax error, unexpected PARSEOP_ARG1
DSDT.dsl 17391: }
Error 6126 - ^ syntax error, unexpected '}'
DSDT.dsl 34118:
Error 6126 - syntax error, unexpected $end and premature End-Of-File
Error 6126 - Compiler aborting due to parser-detected syntax error(s)
DSDT.dsl 13965: Arg1
Error 6126 - ^ syntax error, unexpected PARSEOP_ARG1
DSDT.dsl 17391: }
Error 6126 - ^ syntax error, unexpected '}'
DSDT.dsl 34118:
Error 6126 - syntax error, unexpected $end and premature End-Of-File
Input file: DSDT.dsl - Compilation aborted due to parser-detected syntax error(s)
```
I can't fix it myself, I've tried recipes – there are even more mistakes.

My BIOS
[RebarMODBios.zip](https://github.com/xCuri0/ReBarUEFI/files/14384807/RebarMODBios.zip)
The original BIOS was modified in order 
1) Coffee Time 0.99 + Opening hidden settings sections.
2)  Resize BAR mod.

![Снимок экрана (21)](https://github.com/xCuri0/ReBarUEFI/assets/20264093/501926a9-7ecd-40ac-8029-416fcffaf5d2)
![Снимок экрана (22)](https://github.com/xCuri0/ReBarUEFI/assets/20264093/49aca663-25ce-4fec-9b96-3891c3d0c6c4)
![12](https://github.com/xCuri0/ReBarUEFI/assets/20264093/458b0a57-27da-4de9-97f8-023f51589454)

**Comment by duggasco** [comment]
were you ever able to remedy?
